server/src/main/java/io/crate/execution/dml/upsert/ShardUpsertRequest.java

```diff
@@ -146,6 +152,18 @@ public ShardUpsertRequest(StreamInput in) throws IOException {
         }
     }
 
+    public Operation operation() {
```

Comment:
In plan to use the operation (or type) in the ShardDMLExecutor to figure out what estimate I should take.